### PR TITLE
upgrade docker, weave, marathon

### PR DESCRIPTION
**Pull request by wallies**

### CODE REVIEW

File: packer/ubuntu-14.04_amd64.json [hunk 1]
```diff
@@ -51,6 +56,7 @@
     "guest_os_type": "linux",
     "headless": true,
     "vm_name": "apollo-ubuntu-14.04-amd64-vmware",
+    "name": "apollo-ubuntu-14.04-amd64-vmware",
```
Comment (tayzlor) [inline packer/ubuntu-14.04_amd64.json]: any reason to change this ?

File: packer/ubuntu-14.04_amd64-droplet.json [hunk 2]
```diff
@@ -5,9 +5,9 @@
     "digitalocean_size": "{{env `DIGITALOCEAN_SIZE`}}",
     "digitalocean_image": "{{env `DIGITALOCEAN_IMAGE`}}",
     "mesos_version": "0.22.1-1.0.ubuntu1404",
-    "marathon_version": "0.8.1-1.0.171.ubuntu1404",
-    "consul_version": "0.5.1",
-    "weave_version": "v0.11.2",
+    "marathon_version": "v0.8.2",
```
Comment (tayzlor) [inline packer/ubuntu-14.04_amd64-droplet.json]: changing this version to the docker container version is gonna break the package install

**Comment by enxebre** [comment]
We'll need to rebuild packer images and update artifact version in bootstrap scripts and wercker

File: packer/ubuntu-14.04_amd64-amis.json [hunk 3]
```diff
@@ -3,9 +3,9 @@
     "aws_access_key": "{{env `AWS_ACCESS_KEY_ID`}}",
     "aws_secret_key": "{{env `AWS_ACCESS_KEY`}}",
     "mesos_version": "0.22.1-1.0.ubuntu1404",
-    "marathon_version": "0.8.1-1.0.171.ubuntu1404",
-    "consul_version": "0.5.1",
-    "weave_version": "v0.11.2",
+    "marathon_version": "v0.8.2",
```
Comment (tayzlor) [inline packer/ubuntu-14.04_amd64-amis.json]: 0.8.2-1.0.306.ubuntu1404

File: packer/scripts/ubuntu/install_docker.sh [hunk 4]
```diff
@@ -5,5 +5,13 @@ set -o pipefail
 sudo apt-get -y update
 sudo apt-get install -y linux-image-extra-$(uname -r)
 curl -sSL https://get.docker.com/ubuntu/ | sudo sh
+
+# Download docker containers to the machine to save download time on
+# provisioning later
+docker pull gliderlabs/registrator:master
+docker pull asteris/haproxy-consul:latest
+docker pull weaveworks/weave:${WEAVE_VERSION}
+docker pull mesosphere/marathon:${MARATHON_VERSION}
```
Comment (tayzlor) [inline packer/scripts/ubuntu/install_docker.sh]: hard-code this to pull 0.8.2 for now

**Comment by tayzlor** [comment]
lgtm who's volunteering to rebuild the packer boxes?

File: packer/scripts/ubuntu/install_docker.sh [hunk 5]
```diff
@@ -5,5 +5,13 @@ set -o pipefail
 sudo apt-get -y update
 sudo apt-get install -y linux-image-extra-$(uname -r)
 curl -sSL https://get.docker.com/ubuntu/ | sudo sh
+
+# Download docker containers to the machine to save download time on
+# provisioning later
+docker pull gliderlabs/registrator:master
+docker pull asteris/haproxy-consul:latest
+docker pull weaveworks/weave:${WEAVE_VERSION}
```
Comment (tayzlor) [inline packer/scripts/ubuntu/install_docker.sh]: Can we also pull the weave scope container ?

**Comment by tayzlor** [comment]
Can you create a follow-up ticket to refactor some of the ansible roles where we download / pull stuff - 

docker-py
weavescope

File: roles/haproxy/defaults/main.yml [hunk 6]
```diff
@@ -11,4 +11,4 @@ haproxy_domain: example.com
 consul_template_dir: /mnt/consul-template.d
 consul_template_loglevel: debug
 consul_backend: consul.service.consul:8500
-consul_template_version: 0.8.0
+consul_template_version: 0.9.0
```
Comment (tayzlor) [inline roles/haproxy/defaults/main.yml]: consul template 0.10.0 is out

**Pull request merged**